django/thunderstore/api/cyberstorm/views/packages.py

```diff
@@ -0,0 +1,467 @@
+        cache_vary += f".{self.page_size}"
+        return cache_vary
+
+    def _get_sibling_pages(
+        self,
+        params: OrderedDict,
+        package_page: Page,
+    ) -> Tuple[Optional[str], Optional[str]]:
+        """
+        Return the URLs to previous and next pages of this result set.
+        """
+        for value in params.values():
+            if type(value) == list:
+                value.sort()
+
+        request_url = self.request.get_raw_uri()
+        base_url = request_url.split("?")[0] if "?" in request_url else request_url
+        previous_url = None
+        next_url = None
+
+        if package_page.has_previous():
+            params["page"] -= 1
+            previous_url = f"{base_url}?{urlencode(params, doseq=True)}"
+            params["page"] += 1
+
+        if package_page.has_next():
+            params["page"] += 1
+            next_url = f"{base_url}?{urlencode(params, doseq=True)}"
+            params["page"] -= 1
+
+        return (previous_url, next_url)
+
+
+def get_community_package_queryset(community: Community) -> QuerySet[Package]:
+    """
+    Create base QuerySet for community scoped PackageListAPIViews.
+    """
+    review_statuses = [PackageListingReviewStatus.approved]
+
+    if not community.require_package_listing_approval:
+        review_statuses.append(PackageListingReviewStatus.unreviewed)
+
+    # Ensure each package.community_listings QuerySet will include
+    # only listings related to current community.
+    community_listings = Prefetch(
+        "community_listings",
+        community.package_listings.all(),
+    )
+
+    return (
+        Package.objects.prefetch_related(community_listings)
+        .exclude(~Q(community_listings__review_status__in=review_statuses))
+        .exclude(~Q(community_listings__community__pk=community.pk))
+    )
+
+
+class CommunityPackageListApiView(BasePackageListApiView):
+    """
+    Community-scoped package list.
+    """
+
+    @swagger_auto_schema(
+        query_serializer=PackageListRequestSerializer,
+        responses={200: PackageListResponseSerializer()},
+        operation_id="cyberstorm.package.community",
+        tags=["cyberstorm"],
+    )
+    def get(self, request: Request, *args: Any, **kwargs: Any) -> Response:
+        return super().get(request, *args, **kwargs)
+
+    def _get_base_queryset(self) -> QuerySet[Package]:
+        """
+        Return QuerySet filtered with endpoint specific operations.
+        """
+        community_id = self.kwargs["community_id"]
+        community = get_object_or_404(Community, identifier=community_id)
+
+        return get_community_package_queryset(community)
+
+    def _get_paginator_cache_key(self):
+        return "api.cyberstorm.package.community"
+
+    def _get_paginator_cache_vary_prefix(self):
+        return self.kwargs["community_id"]
+
+
+class TeamPackageListApiView(BasePackageListApiView):
+    """
+    Community & Team-scoped package list.
+    """
```

Comment:
Minor nitpick but this should probably be framed as namespace scoped rather than team scoped. Doesn't matter in practice right now, but we'll have to come back to refactor it in the future if we use team rather than namespace as the scope (and it conflicts with the longer term desired URL structure)